shared/middleware/helpers/sql_table.rb

```diff
@@ -0,0 +1,180 @@
+#
+# SqlTable
+#
+class SqlTable
+  class NotFound < Sinatra::NotFound
+  end
+
+  def initialize(channel_id, storage_id, table_name)
+    _, @channel_id = storage_decrypt_channel_id(channel_id) # TODO(if/when needed): Ensure this is a registered channel?
+    @storage_id = storage_id
+    @table_name = table_name
+
+    @table = PEGASUS_DB[:app_tables]
+    @metadata_table = PEGASUS_DB[:channel_table_metadata]
+  end
+
+  def exists?()
+    !@table.where(app_id: @channel_id, storage_id: @storage_id, table_name: @table_name).limit(1).first.nil?
+  end
+
+  def items()
+    @items ||= @table.where(app_id: @channel_id, storage_id: @storage_id, table_name: @table_name)
+  end
+
+  def metadata
+    dataset = metadata_dataset
+    dataset && dataset.first
+  end
+
+  def metadata_dataset
+    @metadata_dataset ||= @metadata_table.where(app_id: @channel_id, storage_id: @storage_id, table_name: @table_name).limit(1)
+  end
+
+  # create a new metadata row, based on the contents of any existing records
+  def create_metadata
+    @metadata_table.insert({
+      app_id: @channel_id,
+      storage_id: @storage_id,
+      table_name: @table_name,
+      column_list: TableMetadata.generate_column_list(to_a).to_json,
+      updated_at: DateTime.now
+    })
+  end
```

Comment:
If we somehow call create_metadata twice due to some race condition, I think we will end up with a Sequel::UniqueConstraintViolation because the index is marked as unique. Maybe this is worth catching and ignoring?